calico/felix/devices.py

```diff
@@ -95,6 +97,49 @@ def interface_exists(interface):
 
 def list_interface_ips(ip_type, interface):
     """
+    List the local IPs assigned to an interface.
+    :param str ip_type: IP type, either futils.IPV4 or futils.IPV6
+    :param str interface: Interface name
+    :returns: a set of all addresses directly assigned to the device.
+    """
+    if ip_type == futils.IPV4:
+        data = futils.check_call(
+            ["ip", "addr", "list", "dev", interface]).stdout
+        regex = r'^    inet ([0-9.]+)'
```

Comment:
Does the dot need escaping? Otherwise won't this match an arbitrary character sequence?